upup/pkg/fi/cloudup/dotasks/droplet.go

```diff
@@ -210,3 +212,40 @@ func (_ *Droplet) CheckChanges(a, e, changes *Droplet) error {
 	}
 	return nil
 }
+
+type terraformDropletOptions struct {
+	Image    *string                  `cty:"image"`
+	Size     *string                  `cty:"size"`
+	Region   *string                  `cty:"region"`
+	Name     *string                  `cty:"name"`
+	Tags     []string                 `cty:"tags"`
+	SSHKey   []string                 `cty:"ssh_keys"`
+	UserData *terraformWriter.Literal `cty:"user_data"`
+	VPCUUID  *string                  `cty:"vpc_uuid"`
+}
+
+func (_ *Droplet) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *Droplet) error {
+	tf := &terraformDropletOptions{
+		Image:  e.Image,
+		Size:   e.Size,
+		Region: e.Region,
+		Name:   e.Name,
+		Tags:   e.Tags,
+		SSHKey: []string{fi.ValueOf(e.SSHKey)},
+	}
+
+	if e.UserData != nil {
+		d, err := fi.ResourceAsBytes(e.UserData)
+		if err != nil {
+			return err
```

Comment:
Should we wrap this error and the one from line 245 in a `fmt.Errorf` in order to add context distinguishing the two?

Reply:
sure, updated.